### PR TITLE
Feature/7707 - Products And Services page implementation in Account Settings view

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7707

*Description of changes:*

- Products And Services page implementation in Account Settings view.
- The view should handle two scenarios - account creation flow and account settings (edit) flow.
- Different behavior when the product is active/rejected/pending/unsubscribed.
- added test cases for the behavior

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/common/Product.vue [hunk 1]
```diff
@@ -56,6 +64,7 @@
                   v-bind="productFooter.props"
                   v-on="productFooter.events"
                   :ref="productFooter.ref"
+                  v-display-mode="hasDecisionNotBeenMade ?'': 'VIEW_ONLY'"
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/Product.vue:67]: re-use VIEW_ONLY from constant file

File: auth-web/src/components/auth/common/Product.vue [hunk 2]
```diff
@@ -103,8 +115,17 @@ export default class Product extends Vue {
     // lang file have subtitle and description with product code prefix.
     // eg: pprCodeSubtitle, pprCodeDescription
     const { code } = this.productDetails
-    const subTitle = `${code && code.toLowerCase()}CodeSubtitle` || ''
-    const details = `${code && code.toLowerCase()}CodeDescription` || ''
+    let subTitle = `${code && code.toLowerCase()}CodeSubtitle` || ''
+    let details = `${code && code.toLowerCase()}CodeDescription` || ''
+    if (this.isAccountSettingsView && this.productDetails.subscriptionStatus === productStatus.ACTIVE) {
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/Product.vue:120]: better to use switch?

File: auth-web/src/components/auth/common/Product.vue (hunk 1 above)
Comment (karthik-aot) [reply auth-web/src/components/auth/common/Product.vue:67]: yeah. refactored :)

File: auth-web/src/components/auth/common/Product.vue (hunk 2 above)
Comment (karthik-aot) [reply auth-web/src/components/auth/common/Product.vue:121]: good idea. 👍

**Review by saravanpa-aot — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

**Review by shabeeb-aot — APPROVED** [review]

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**